sycl/include/sycl/ext/oneapi/experimental/invoke_simd.hpp

```diff
@@ -349,6 +362,8 @@ __attribute__((always_inline)) auto invoke_simd(sycl::sub_group sg,
   // is fine in this case.
   constexpr int N = detail::get_sg_size<Callable, T...>();
   using RetSpmd = detail::SpmdRetType<N, Callable, T...>;
+  detail::verify_return_type_matches_sg_size<
```

Comment:
The arguments need to pass similar check:
if sg-size is N, and one of operands is simd<T, N*2>, then it is the reportable error as well.

Reply:
Yeah I originally implemented that, but it turns out we already check for this when picking a subgroup size.

Example:

```
void SIMD_CALLEE( simd<float, VL> b,
                               simd<float, VL*2> c )  SYCL_ESIMD_FUNCTION {
```

Error:
```
error: static assertion failed due to requirement '__MP11_NS::integral_constant<unsigned long, 0>::value == 1'
    static_assert((__MP11_NS::mp_size<InvocableSgSizes>::value == 1) &&

```

We didn't do the return type check there because we need the finalized subgroup size for the return type.

Reply:
I see, thank you for explanations.
The comment on the line 361 concerns me a bit.
It says 0 is fine. If N is 0 (i.e. all parameters are uniform), then verify_return_type_matches_sg_size would fail, right? If Yes, then this special case needs to be handled specially in the static_assert that was added in this PR.